### PR TITLE
adding hover behavior for images

### PR DESCRIPTION
**Pull request by mirnawong1**
AI Summary: This pull request primarily involves changes to the image display size in various markdown files. The changes reduce the width of images displayed using the `<Lightbox>` tag from the original size to a smaller size, typically 90%. This adjustment is likely to improve the layout and readability of the documentation pages. 

this pr adds a 'hover over images to expand' behavior. the following changes were made to the original code:

## original lightbox src component: 
- original lightbox component handled the display of images with alignment and collapsing features.
- used props for customization but did not have any interactive behavior like hover effects or handling of scrolling events.

## updates to lightbox src component:
- added state variables : `isHovered`, `expandImage`, and `isScrolling`.
- `isHovered`: tracks if the mouse is hovering over image.
- `expandImage`: identifies if the image should expand.
- `isScrolling`: helps indicate whether page is currently being scrolled. (to avoid pop ups when a user is reading)
- implemented a hover effect that expands the image (`setExpandImage(true)` after a delay (300 milliseconds), but only if the page is not being scrolled (!isScrolling).
- added a `handleScroll` function to update the `isScrolling` state when the user scrolls. This function also cancels the image expansion (`setExpandImage(false)`) and resets `isScrolling` after a delay.
- added event listeners for mouse enter (`onMouseEnter`) and mouse leave (`onMouseLeave`) to control the hover state (`setIsHovered`).
- attached a scroll event listener to the window in a `useEffect` hook to trigger `handleScroll`
- modified the `handleMouseEnter` to include a delay (matching the scroll delay) before setting `isHovered` to true. This ensures that hover effects are not activated immediately after scrolling stops.
- adjusted the component structure and styling to accommodate the new hover and scroll behaviors. this means the image now scales (transform: 'scale(1.2)') with a smooth transition (transition: 'transform 0.5s ease') when hovered, and the zIndex is increased to '9999' during expansion.
- added a default width to `80%` to all images (using lightbox src component) to ensure the scale was consistent across images and improve accessibility. adding widths also helps with page load and SEO. 

## images
- also adjusted widths for images that a large width attribute assigned. this is to ensure it scaled correctly without affecting the user UX.

### CODE REVIEW

File: website/blog/2023-04-17-dbt-squared.md [hunk 1]
```diff
@@ -50,11 +50,11 @@ We needed a way to make this architecture manageable when dozens of downstream t
 
 The second architectural decision was whether or not to create a single dbt project for all 50+ country teams, or to follow a multi-project approach in which each country would have its own separate dbt project in the shared repo. It was critical that each country team was able to move at different paces and have full control over their domains. This would avoid issues like model name collisions across countries and remove dependencies that would risk cascading errors between countries. Therefore, we opted for a one project per country approach.
 
-<Lightbox src="/img/blog/2023-04-17-dbt-squared/roche-project-tree.png" width="85%" title="Roche project structure as seen in the repository" />
+<Lightbox src="/img/blog/2023-04-17-dbt-squared/roche-project-tree.png" width="65%" width="85%" title="Roche project structure as seen in the repository" />
```
Comment (mirnawong1) [inline website/blog/2023-04-17-dbt-squared.md]:
```suggestion
<Lightbox src="/img/blog/2023-04-17-dbt-squared/roche-project-tree.png" width="85%" title="Roche project structure as seen in the repository" />
```

File: website/src/components/lightbox/styles.module.css [hunk 2]
```diff
@@ -24,3 +24,9 @@
 .rightAlignLightbox {
   margin: 10px 0 10px auto;
 }
+
+:local(.hovered) { /* Add the . before the class name */
```
Comment (mirnawong1) [inline website/src/components/lightbox/styles.module.css]:
```suggestion
:local(.hovered) {
```

File: website/blog/2023-04-18-building-a-kimball-dimensional-model-with-dbt.md [hunk 3]
```diff
@@ -185,7 +185,7 @@ Now that you’ve set up the dbt project, database, and have taken a peek at the
 
 Identifying the business process is done in collaboration with the business user. The business user has context around the business objectives and business processes, and can provide you with that information. 
 
-<Lightbox src="/img/blog/2023-04-18-building-a-kimball-dimensional-model-with-dbt/conversation.png" width="65%" title="Conversation between business user and analytics engineer"/>
+<Lightbox src="/img/blog/2023-04-18-building-a-kimball-dimensional-model-with-dbt/conversation.png" title="Conversation between business user and analytics engineer"/>
```
Comment (mirnawong1) [inline website/blog/2023-04-18-building-a-kimball-dimensional-model-with-dbt.md:188]: the original images don't have a width attribute and this undo's a python script i ran, which added a width attribute.

**Comment by mirnawong1** [comment]
also not sure if the scrolling is affecting the hover a little

File: website/src/components/lightbox/index.js [hunk 4]
```diff
@@ -1,34 +1,65 @@
...
-      <span 
+      <div
         className={`
-          ${styles.docImage} 
+          ${styles.docImage}
           ${collapsed ? styles.collapsed : ''}
-          ${imageAlignment}
+          ${alignment === "left" ? styles.leftAlignLightbox : ''}
+          ${alignment === "right" ? styles.rightAlignLightbox : ''}
+          ${isHovered ? styles.hovered : ''}
```
Comment (JKarlavige) [inline website/src/components/lightbox/index.js:58]: Nice work with these conditional checks!

**Review by JKarlavige — APPROVED** [review]
The scale on hover looks like it works great, awesome work @mirnawong1!

**Pull request merged**